File: backend/src/main/java/com/people_manager/dto/response/EmployeeResponseDto.java
package com.people_manager.dto.response;

import com.people_manager.enums.EmployeeStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class EmployeeResponseDto {
    private Long id;
    private String name;
    private LocalDate hireDate;
    private BigDecimal salary;
    private EmployeeStatus status;
}

File: frontend/src/models/employee/EmployeeResponseDto.js
/**
 * DTO para resposta de funcionário
 */
export class EmployeeResponseDto {
    constructor(id, name, hireDate, salary, status, createdAt, updatedAt) {
        this.id = id;
        this.name = name;
        this.hireDate = hireDate;
        this.salary = salary;
        this.status = status;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
}
